src/tools/PublicAnalyzers/AvaloniaPropertyAnalyzer.cs

```diff
@@ -0,0 +1,463 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Collections.ObjectModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Runtime.Serialization;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Avalonia.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+[SuppressMessage("MicrosoftCodeAnalysisReleaseTracking", "RS2008:Enable analyzer release tracking")]
+public partial class AvaloniaPropertyAnalyzer : DiagnosticAnalyzer
+{
+    private const string Category = "AvaloniaProperty";
+
+    private const string TypeMismatchTag = "TypeMismatch";
+    private const string NameCollisionTag = "NameCollision";
+    private const string AssociatedClrPropertyTag = "AssociatedClrProperty";
+    private const string InappropriateReadWriteTag = "InappropriateReadWrite";
+
+    private static readonly DiagnosticDescriptor AssociatedAvaloniaProperty = new(
+        "AVP0001",
+        "Identification of the AvaloniaProperty associated with a CLR property",
+        "Associated AvaloniaProperty: {0} {1}",
+        Category,
+        DiagnosticSeverity.Info,
+        isEnabledByDefault: false,
+        "This informational diagnostic identifies which AvaloniaProperty a CLR property is associated with.",
+        AssociatedClrPropertyTag);
+
+    private static readonly DiagnosticDescriptor InappropriatePropertyAssignment = new(
+        "AVP1000",
+        "AvaloniaProperty objects should be stored appropriately",
+        "Incorrect AvaloniaProperty storage: {0} should be static and readonly",
+        Category,
+        DiagnosticSeverity.Warning,
+        isEnabledByDefault: true,
+        "AvaloniaProperty objects have static lifetimes and should be stored accordingly. Do not multiply construct the same property.");
```

Comment:
"multiply construct" doesn't sound grammatically correct to me. I'm not sure what you mean here either, you mean they should be static so multiple instances aren't created per control?

Reply:
Yes, that's the meaning. "Multiply" is an adverb here, which is a valid but perhaps obscure usage. Sounds like the message should be rewritten.

Reply:
In fact this whole sentence is unnecessary, because the very next diagnostic handles multiple construction.